### ISSUE TITLE
Optional shuffling for DistributedSampler

### ISSUE THREAD

**Issue by erogol** [pytorch/pytorch#14474]
## 🚀 Feature
By default DistributedSampler shuffles the dataset. It would be better to make shuffling optional.

## Motivation

In some certain cases, it is better to disable it. For instance in seq2seq models where you better to sort instances based on sequence length.

## Pitch

We could introduce a new argument ```shuffle=True or False``` or it might be passed from the dataloader.

**Comment by f-sky** [comment]
You can visit my repo and find one.
https://github.com/f-sky/clh_utils/blob/master/clh_utils/pytorch_ext/sampler.py

**Comment by apsdehal** [comment]
This can be closed after #22479.
